webui/react/src/components/ColumnPickerMenu.tsx

```diff
@@ -95,18 +99,28 @@ const ColumnPickerTab: React.FC<ColumnTabProps> = ({
   }, [searchString, totalColumns, tab]);
 
   const allFilteredColumnsChecked = useMemo(() => {
-    return filteredColumns.every((col) => columnState.includes(col.column));
+    return filteredColumns.every((col) => columnState.includes(formatColumnKey(col)));
   }, [columnState, filteredColumns]);
 
   const handleShowHideAll = useCallback(() => {
-    const filteredColumnMap: Record<string, boolean> = filteredColumns.reduce(
-      (acc, col) => ({ ...acc, [col.column]: columnState.includes(col.column) }),
-      {},
-    );
+    const filteredColumnMap: Record<string, boolean> = filteredColumns.reduce((acc, col) => {
+      if (col.location === V1LocationType.RUNMETADATA)
+        return { ...acc, [formatColumnKey(col)]: columnState.includes(formatColumnKey(col)) };
+
+      return { ...acc, [col.column]: columnState.includes(col.column) };
```

Comment:
see prev re: `formatColumnKey`
```suggestion
      return { ...acc, [formatColumnKey(col)]: columnState.includes(col.column) };
```